### PR TITLE
UV backend HTTP probe

### PR DESCRIPTION
**Pull request by eea03**
- implemented backend service HTTP probe (embedded Jetty HTTP server)
- using same code as frontend probe (check to database)
- by default runs on port 8066 (can be modified in config.properties)

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -54,6 +54,11 @@
 			<version>${mockito.version}</version>
 			<scope>test</scope>
 		</dependency>
+        <dependency>
```
Comment (skrchnavy) [inline backend/pom.xml]: Please reformat the file (using spaces)

**Comment by tomas-knap** [comment]
Under which Java did you test it? Under Java 8?

The introduced dependency - jetty-util-9.3.3.v20150827.jar - seems to be compiled under Java 8. So when starting backend under Java 7 (which is still the main target Java version), it throws exception:

```
java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/AbstractHandler : Unsupported major.minor version 52.0
```

Full exception:

```
015-09-21 13:33:39,749 [main] INFO  exec: dpu: o.e.persistence.logging.weaver - 2015-09-21 13:33:39.749--ServerSession(120821458)--Class cz.cuni.mff.xrg.odcs.commons.app.user.RoleEntity could not be weaved for change tracking as it is not supported by its mappings.
Exception in thread "main" org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'moduleFacade': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: private cz.cuni.mff.xrg.odcs.commons.app.module.osgi.OSGIModuleFacadeConfig cz.cuni.mff.xrg.odcs.commons.app.module.osgi.OSGIModuleFacade.configuration; nested exception is org.springframework.beans.factory.CannotLoadBeanClassException: Error loading class [cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler] for bean with name 'cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler#0' defined in class path resource [backend-context.xml]: problem with class file or dependent class; nested exception is java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/AbstractHandler : Unsupported major.minor version 52.0
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:287)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1106)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:456)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:294)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:225)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:291)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:607)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:925)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:472)
	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:139)
	at org.springframework.context.support.ClassPathXmlApplicationContext.<init>(ClassPathXmlApplicationContext.java:83)
	at cz.cuni.mff.xrg.odcs.backend.AppEntry.initSpring(AppEntry.java:76)
	at cz.cuni.mff.xrg.odcs.backend.AppEntry.run(AppEntry.java:220)
	at cz.cuni.mff.xrg.odcs.backend.AppEntry.main(AppEntry.java:269)
Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private cz.cuni.mff.xrg.odcs.commons.app.module.osgi.OSGIModuleFacadeConfig cz.cuni.mff.xrg.odcs.commons.app.module.osgi.OSGIModuleFacade.configuration; nested exception is org.springframework.beans.factory.CannotLoadBeanClassException: Error loading class [cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler] for bean with name 'cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler#0' defined in class path resource [backend-context.xml]: problem with class file or dependent class; nested exception is java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/AbstractHandler : Unsupported major.minor version 52.0
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:513)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:92)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:284)
	... 15 more
Caused by: org.springframework.beans.factory.CannotLoadBeanClassException: Error loading class [cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler] for bean with name 'cz.cuni.mff.xrg.odcs.backend.communication.ProbeHandler#0' defined in class path resource [backend-context.xml]: problem with class file or dependent class; nested exception is java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/AbstractHandler : Unsupported major.minor version 52.0
	at org.springframework.beans.factory.support.AbstractBeanFactory.resolveBeanClass(AbstractBeanFactory.java:1265)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:576)
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1331)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doGetBeanNamesForType(DefaultListableBeanFactory.java:337)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanNamesForType(DefaultListableBeanFactory.java:316)
	at org.springframework.beans.factory.BeanFactoryUtils.beanNamesForTypeIncludingAncestors(BeanFactoryUtils.java:187)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:856)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:813)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:730)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:485)
	... 17 more
Caused by: java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/AbstractHandler : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:791)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:423)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:356)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:791)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:423)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:356)
	at org.springframework.util.ClassUtils.forName(ClassUtils.java:258)
	at org.springframework.beans.factory.support.AbstractBeanDefinition.resolveBeanClass(AbstractBeanDefinition.java:417)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doResolveBeanClass(AbstractBeanFactory.java:1283)
	at org.springframework.beans.factory.support.AbstractBeanFactory.resolveBeanClass(AbstractBeanFactory.java:1254)
	... 26 more
```

**Comment by eea03** [comment]
You are right, for some reason I run backend under Java 8.
I fixed the dependency - last released version compiled under Java 7.
I tested it and it runs ok now both with Java 7 and Java 8.

**Comment by tomas-knap** [comment]
Checked, working correctly.

**Comment by skrchnavy** [comment]
cherrypicked into release UV_Core_2.2.0 branch in commits:
6bf2ce4d63b38177241b7a802e9ec02e1686b940 - 511dc02237f902651753d77a1bbf61b009a3d9e6

**Pull request merged**